app/src/main/kotlin/com/wire/android/util/FileUtil.kt

```diff
@@ -104,25 +104,26 @@ private fun Context.saveFileDataToDownloadsFolder(assetName: String, downloadedD
             put(MIME_TYPE, mimeType)
             put(SIZE, fileSize)
         }
-        resolver.insert(MediaStore.Downloads.EXTERNAL_CONTENT_URI, contentValues)
+        resolver.insert(MediaStore.Downloads.EXTERNAL_CONTENT_URI, contentValues)?.also { downloadedUri ->
+            resolver.copyFile(downloadedUri, downloadedDataPath)
+        }
     } else {
         val authority = getProviderAuthority()
         val destinationFile = File(downloadsDir, availableAssetName)
-        val uri = FileProvider.getUriForFile(this, authority, destinationFile)
-        if (mimeType?.isNotEmpty() == true) {
-            val downloadManager = getSystemService(Context.DOWNLOAD_SERVICE) as DownloadManager
+        val downloadManager = getSystemService(Context.DOWNLOAD_SERVICE) as DownloadManager
+        FileProvider.getUriForFile(this, authority, destinationFile).also { downloadedUri ->
+            resolver.copyFile(downloadedUri, downloadedDataPath)
             downloadManager.addCompletedDownload(
                 /* title = */ availableAssetName,
                 /* description = */ availableAssetName,
                 /* isMediaScannerScannable = */ true,
-                /* mimeType = */ mimeType,
+                /* mimeType = */ mimeType ?: "*/*",
```

Comment:
looking at the iOS code there is a chance that the mimeType is an empty string "" so maybe we can check for that here

Reply:
Good idea! Added that check to the code